### PR TITLE
configurable postpone

### PR DESCRIPTION
**Pull request by LukaszWiktor**
I've made the postpone time configurable.

image

image

Additionally, I've fixed a bug with setting **breaks to** - the grabTime function was only working for **breaks from**.

### CODE REVIEW

File: templates/settings.html [hunk 1]
```diff
@@ -36,10 +36,15 @@
                                     ng-model="config.frequency" required/> <span>minutes</span>
                             </div>
                             <div class="form-group" ng-show="config.notificationType === 'F'">
-                                <label for="break-frequency">Break length</label>
+                                <label for="break-length">Break length</label>
                                 <input type="number" id="break-length" class="form-control min-input"
                                     ng-model="config.length" required/> <span>minutes</span>
                             </div>
+                            <div class="form-group" ng-show="config.notificationType === 'F'">
+                                <label for="postpone">Postpone</label>
```
Comment (tom-james-watson) [inline templates/settings.html]: I'd have the label say "Postpone Length" (and the id "postpone-length" for consistency)

File: js/background.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ var fullscreenSetInterval;
 var defaultConfig = {
     frequency: 28,
     length: 2,
+    postpone: 3,
```
Comment (tom-james-watson) [inline js/background.js:13]: I haven't had chance to test this yet, but I think that the settings are persistent across version updates, which means that when a user receives this update, config.postpone would be undefined when we fetch their settings from local storage.

I think we need to update https://github.com/LukaszWiktor/breaktimer/blob/7a68d93d830874353e0c76a28c541ee6032188e4/js/background.js#L52 to something like

```
// Grab config from local storage or take defaultConfig
chrome.storage.local.get('config', function(data) {
    if (typeof(data.config) === 'undefined') {
        setConfig(defaultConfig);
    } else {
        if (typeof(data.config.postpone) === 'undefined') {
            data.config.postpone = defaultConfig.postpone;
        }
        setConfig(data.config);
    }
});
```

Not ideal, but I can't think of a better way of doing this.

**Comment by tom-james-watson** [comment]
All in all this looks amazing! Thank you!

File: templates/settings.html (hunk 1 above)
Comment (LukaszWiktor) [reply templates/settings.html]: Ok, I'll change it.

File: js/background.js (hunk 2 above)
Comment (LukaszWiktor) [reply js/background.js:13]: Right, I haven't think about it.

**Comment by LukaszWiktor** [comment]
@tom-james-watson No big deal. It was a pleasure to work with your code. I'll update the PR in the evening.

**Comment by LukaszWiktor** [comment]
Hey @tom-james-watson. I updated the PR. I solved the issue with missing properties in config from previous versions by merging it with default config:

```
var config = Object.assign({}, defaultConfig, data.config);
setConfig(config);
```

For fresh installations it will take all defaults.

File: js/background.js [hunk 3]
```diff
@@ -48,13 +49,10 @@ var defaultConfig = {
...
 
-// Grab config from local storage or take defaultConfig
+// Grab config from local storage mergded with defaultConfig
 chrome.storage.local.get('config', function(data) {
-    if (typeof(data.config) === 'undefined') {
-        setConfig(defaultConfig);
-    } else {
-        setConfig(data.config);
-    }
+    var config = Object.assign({}, defaultConfig, data.config);
```
Comment (tom-james-watson) [inline js/background.js:54]: Great solution!

**Review by tom-james-watson — APPROVED** [review]

**Comment by tom-james-watson** [comment]
I've uploaded the new version to the chrome store - you should get the update in the next few hours.

Thank you very much for your contribution!

**Comment by LukaszWiktor** [comment]
I've got the update. Thanks for merging my PR and publishing the new version, and first of all, thank you very much for creating the Break Timer extension. It helps me a lot to take regular breaks.

**Pull request merged**